### ISSUE TITLE
AvatarGroup: Add Accessibility section to docs

### ISSUE THREAD

**Issue by sopranopillow** [microsoft/fluentui#24699]
Add an Accessibility section to AvatarGroup similar to spinbutton's: https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-spinbutton/src/stories/SpinButton/SpinButtonAccessibility.md.

This issue should be referenced as it's a known issue: #24471 

Related: #22240, #23773, #24696, #24697
